docs/ProblemStatementAndGoals/ProblemStatement.tex

```diff
@@ -47,31 +47,107 @@ \subsection{Inputs and Outputs}
 
 \section{Goals}
 
+\begin{enumerate}
+    \item \textbf{TPG Framework must adhere to a modern software engineering Dev Ops pipeline}
+    
+    Set up fully automated CI pipeline to ensure all code changes in the TPG framework are continuously tested, verified and follow proper guidelines
+
+
+\textbf{Measurable Outcomes: 
+} To accomplish this goal, there should be a complete integration of a CI tool such as Github Actions or Gitlab CI. The CI pipeline should be invoked when changes are made to the framework via new PRs and should not result in blocking errors (either accept or reject changes after CI pipeline analyzes changes). 
+
+\textbf{Reasoning: } Establishing a CI pipeline will streamline the development process, reduce human error, and improve code reliability. Automating testing ensures that the framework is stable with each update, promoting scalability and maintainability
+
+\item \textbf{ The TPG Framework should be well tested}
+
+Develop a comprehensive testing suite for the TPG framework, including unit tests, to ensure major components are thoroughly validated and code coverage is maximized.
+
+\textbf{Measurable Outcomes:
+}To accomplish this goal, unit and integration tests will be implemented for core TPG functions, ideally hitting at least 80\% code coverage (subject to change after further analysis). The tests must be able to be run as part of the CI pipeline, ensuring that all functionality remains valid after code changes.
+
+
+\textbf{Reasoning:} Introducing a well-structured testing suite is critical for ensuring the reliability and correctness of the TPG framework. Unit tests will verify that individual components behave as expected, and adding integration tests will confirm the system functions correctly as a whole. High test coverage ensures that potential issues are caught early, reducing future bugs and improving maintainability. This goal also aligns with the project’s emphasis on adopting best practices in software engineering.
+
+\item \textbf{The TPG framework must have an Agent-Environment Interface between itself and MuJoCo}
+
+Develop a robust interface that allows the TPG framework to communicate and control robotic agents within the MuJoCo physics engine.
+
+\textbf{Measurable Outcomes:
+} To accomplish this goal,  the TPG framework should contain a fully functional interface for a basic agent (e.g., a simple robotic arm) in MuJoCo. By the end of the project, this can be further expanded to having the interface to support more complex agents, such as quadrupeds.
+
+\textbf{Reasoning:} This interface will enable the TPG framework to be tested in more realistic and complex environments. Successful integration with MuJoCo will validate TPG’s performance in a cutting-edge robotics simulator, essential for showcasing its real-world capabilities.
+
+
+\item \textbf{The TPG Integration in MuJoCo should have determinable behavior when ran}
```

Comment:
I think I'd reword this goal

Some of the problems Dr. Kelly said we'd encounter are:
- Temporal Credit Assignment Problem (which is that sometimes things go bad first then become better)
- Agents need to have memory
- Current testing is in stationary environments, however MuJoCo enables testing in dynamic environments
- TPG is mean't to be an alternative to traditional Deep Learning NN's, maybe we can do an evaluation between TPG vs a more traditional DL NN in the MuJoCo environment to determine which is more performant? I know that the main draw to TPG is the cost, but can it outperform the alternative?
- Kelly wants to explore embedded evolution, could there be a goal which encompasses this?